src/Stryker.Core/Stryker.Core/Initialisation/ProjectFileReader.cs

```diff
@@ -107,21 +107,29 @@ private IAnalyzerResult SelectAnalyzerResult(IAnalyzerResults analyzerResults, s
                 return analyzerResults.First(e => e.TargetFramework != null);
             }
 
-            var analyzerResultForFramework = analyzerResults.SingleOrDefault(result => result.TargetFramework == targetFramework);
+            var analyzerResultList = analyzerResults?.Where(result => result.TargetFramework == targetFramework).ToList();
+
+            switch (analyzerResultList?.Count)
+            {
+                case 0:
+                    throw new InputException($"The target framework of {targetFramework} has no analyzer results.");
+                case > 1:
```

Comment:
So what do you need to have in your `<TargetFrameworks>` to hit this case?

Reply:
> I am afraid you somehow broke the logic: it seems integration tests are now failing raising errors on project analysis, see [logs here](https://dev.azure.com/stryker-mutator/Stryker/_build/results?buildId=9811&view=logs&j=99031f74-bf3d-58da-997c-4e095f08dc64&t=b79c55ad-f33a-5882-eb36-5a64c1e39811&l=1335). Can you fix your PR and make sure integration tests are passing?

I made a modification to this after @dupdob indicated there were failing integration tests. Hopefully, this passes. There are several ways to hit this, but as of right now I'll just start with the fact that analyzerResults can have no results. This causes an issue when we do First() in the IF statement. Likewise, you can pass in a target framework that doesn't match and I wasn't getting a validation error (should there be validation indicating if I flubbed a framework name, like nett6 as a typo example?

Reply:
I did try passing in net472 as an option. I've tried others. I don't see the framework options listed on the site, so I've assumed that the options are identical to the MS listed options? Likewise, I can see that it is picking up my project, but it gets No analyzer results. I'm obviously newer to the framework but was hoping to get this up and running to show the value add. Am I wrong in thinking that this project supports .NET  Framework? 

Based on [this doc](https://stryker-mutator.io/docs/stryker-net/getting-started#dotnet-framework-specific) I assume it is supported. Likewise, I know it isn't my nuget registry as I see the nuget requests in the logs when I have a trace level set for the logging.

Reply:
I'll start looking there then. Certainly, I have an edge case of MSBuild. I'm still desiring to have this project provide more information about the condition that was hit, if possible. It makes me wonder less about why I'm seeing something that I believe shouldn't be exceptional.

Reply:
I believe we can enable debug logging for buildalyzer. We have not enabled this yet because it logs the complete build output of MSBuild but perhaps we should include this as part of trace logging so we can better understand these cases. You could enable this on your local project: https://github.com/daveaglick/Buildalyzer#logging